src/app/CommandHandler.cpp

```diff
@@ -64,7 +64,20 @@ CHIP_ERROR CommandHandler::AllocateBuffer()
         mCommandMessageWriter.Init(std::move(commandPacket));
         ReturnErrorOnFailure(mInvokeResponseBuilder.InitWithEndBufferReserved(&mCommandMessageWriter));
 
-        mInvokeResponseBuilder.SuppressResponse(mSuppressResponse);
+        if (mReserveSpaceForMoreChunkMessages)
+        {
+            ReturnErrorOnFailure(mInvokeResponseBuilder.ReserveSpaceForMoreChunkedMessages());
+        }
+
+        // Sending an InvokeResponse to a InvokeResponse is going to be removed from spec soon.
+        // It was never implemented in the SDK, and there are no commands responses that expect a
+        // command response. This means we will never recieve an InvokeResponse Message in response
+        // to an InvokeResponse Message that we are sending out. This means that the only response
+        // we are expecting to receive in responses to an InvokeResponse Message that we are
+        // sending-out is a status when we are chunking multiple responses. As a result, to satisfy the
+        // condition that we don't set SuppressResponse to true while also setting
+        // MoreChunkedMessages to true, we are hardcoding the value to false here.
+        mInvokeResponseBuilder.SuppressResponse(/* aSuppressResponse = */ false);
```

Comment:
We should remove mSuppressResponse, right?

Reply:
Not really as we should be listening to it if we are only sending back commands. But that is outside the scope of what I am trying to do here